packages/ui/src/components/notifications/NotificationCenter.test.tsx

```diff
@@ -0,0 +1,169 @@
+import { act, fireEvent, render, waitFor } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+
+import NotificationCenter, { notify } from './NotificationCenter';
+import { Notification } from './types';
+
+type SnackbarProps = Notification & { onClose: (id: string) => void };
+
+function randomString() {
+    return Math.random().toString(36).substring(2, 15);
+}
+const mockRandomUUID = vi.fn(() => {
+    return `${randomString()}-${randomString()}-${randomString()}-${randomString()}-${randomString()}`;
+});
+global.crypto = {
+    subtle: global.crypto.subtle,
+    getRandomValues: vi.fn(),
+    randomUUID: mockRandomUUID as () => `${string}-${string}-${string}-${string}-${string}`,
+};
+const NOTIFICATION_EVENT = 'emit-core-notification';
+
+describe('NotificationCenter', () => {
+    const mockNotification = (message: string) => {
+        const notification: Partial<Notification> = {
+            type: 'info',
+            message,
+            createdAt: new Date(),
+            id: randomString(),
+        };
+        return notification;
+    };
+    const emitEvent = (message: string) => {
+        const event: CustomEvent<Partial<Notification>> = new CustomEvent(NOTIFICATION_EVENT, {
+            detail: mockNotification(message),
+        });
+        fireEvent(document, event);
+    };
+    const SnackComponent = ({ onClose, id, message }: SnackbarProps) => (
+        <button type="button" onClick={() => onClose(id)} key={id}>
+            {message}
+        </button>
+    );
+    const renderComponent = (props: { hideAfter?: number; maxNotifications?: number; cacheTimeout?: number } = {}) => {
+        return render(<NotificationCenter SnackbarComponent={SnackComponent} {...props} />);
+    };
+
+    beforeEach(() => {
+        vi.useFakeTimers();
+    });
+
+    it('should render a notification', () => {
+        const { getByText } = renderComponent();
+
+        emitEvent('test');
+
+        expect(getByText('test')).toBeInTheDocument();
+    });
+
+    it("should render a stack of notifications, up to the component's maxNotifications prop", () => {
+        const { queryByText } = renderComponent();
+
+        waitFor(() => {
```

Comment:
`waitFor` retries if something inside it throws an error, and it should always be used with await in front of it. It's usually used with `expect` calls inside that throw if they fail in order to wait for re-renders or updates. The events can be emitted here all at once without `waitFor`